markdown.go

```diff
@@ -314,24 +319,20 @@ func MarkdownBasic(input []byte) []byte {
 //
 // * Custom Heading IDs
 func MarkdownCommon(input []byte) []byte {
-	// set up the HTML renderer
-	renderer := NewHTMLRenderer(HTMLRendererParameters{
-		Flags: CommonHTMLFlags,
-	})
-	return Markdown(input, renderer, DefaultOptions)
+	return Markdown(input, DefaultOptions)
 }
 
 // Markdown is the main rendering function.
 // It parses and renders a block of markdown-encoded text.
-// The supplied Renderer is used to format the output, and extensions dictates
-// which non-standard extensions are enabled.
+// The supplied options contain a Renderer used to format the output, and
+// extensions that dictate which non-standard extensions are enabled.
```

Comment:
Yeah, that's a fundamental problem of trying to provide a set of sane defaults: how do you communicate what the zero does: is it the zero customization or is it the default options?

The current design (and I didn't change it with this PR) intends users to call the convenience functions `MarkdownBasic` and `MarkdownCommon` to distinguish between the zero options and default options. This might, or might not be easy to grasp from the docs; we could explicitly describe  the three invocations in the package-level section.

`DefaultOptions` is another helper to distinguish that, but this one, too, might slip the documentation reader's attention.

But our API surface is very small, maybe we're overthinking this bit :-)